l10n_ch_invoice_reports/models/mail_template.py

```diff
@@ -0,0 +1,61 @@
+# Copyright 2021 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+
+import base64
+
+from odoo import _, models
+
+
+class MailTemplate(models.Model):
+    _inherit = "mail.template"
+
+    def generate_email(self, res_ids, fields=None):
+        """ Method overridden in order remove QR/ISR payslips and use
+            Invoice report with payslip generated by this module
+        """
+        self = self.with_context(_mail_template_no_attachments=True)
```

Comment:
Could this function be simplify by first checking if the model is "account.move" or not ?

Reply:
Agree. Also this seems wrong because you are setting this ctx key every time.
We should set it - if possible - where the email generation is triggered.

Reply:
@simahawk @TDu  reworked method and overwritten ctx in action for that, should be better now.